contracts/core/Vault.sol

```diff
@@ -127,7 +127,6 @@ contract Vault is
     )
         external
         onlyAuthorized
-        isNonZero(_quantity)
```

Comment:
Why are these being removed?

Reply:
Because to we might batch deposit 0 tokens from one exchange but we still are passing on an array that contains the 0 so needed to remove that. Unless we want to do some array pruning somewhere.

Reply:
Can you give me an example? I think this is suboptimal for gas, prefer some kind of efficient pruning